perf/src/main/scala/perf/CompareMetadata.scala

```diff
@@ -0,0 +1,112 @@
+package perf
+
+import better.files.File
+import io.circe
+import io.circe.generic.auto._
+import io.circe.parser._
+import cats.implicits._
+import java.time.Duration
+
+import cats.data.Validated.{Invalid, Valid}
+import common.validation.ErrorOr._
+
+// Do not remove this unused import
+// If removed, circe parser fails to find implicit decoder for OffsetDateTime, and parsing fails
+import io.circe.java8.time.decodeOffsetDateTimeDefault
+
+object CompareMetadata extends App {
+
+  def parseMetadataFromFile(filePath: String): Either[circe.Error, Metadata] = {
+    val metadataFile = File(filePath)
+    val metadataFileContent = metadataFile.contentAsString
+
+    decode[Metadata](metadataFileContent)
+  }
+
+
+  def displayComputedMetrics(metadata: Metadata, displayMsg: String): Unit = {
+    println(displayMsg)
+
+    println(s"Workflow started after: ${metadata.workflowStartedAfter}")
+    println(s"Workflow Running time: ${metadata.workflowRunningTime}")
+    println(s"Total jobs per root workflow: ${metadata.totalJobsPerRootWf}")
+    println(s"Avg cache copy retries: ${metadata.avgCacheRetries}")
+    println(s"Avg time job spent in Call Caching state: ${metadata.avgTimeInCallCachingState}")
+    println(s"Avg time job spent in Job Preparation state: ${metadata.avgTimeInJobPreparation}")
+    println(s"Avg time job spent in fetching and copying cache hit(s) state: ${metadata.avgTimeForFetchingAndCopyingCacheHit}")
+  }
+
+  def compareDurationMetrics(metadataOld: Metadata, metadataNew: Metadata, metricFunc: Metadata => Duration, metricName: String): ErrorOr[String] = {
+    if(metricFunc(metadataNew).compareTo(metricFunc(metadataOld).multipliedBy(1.1.toLong)) > 0){
```

Comment:
Re: `1.1.toLong` This is the same as `1L`, so it doesn't actually multiply the amount by anything.

Instead, you should convert each duration `.toMillis`, multiply the `Old` by `1.1`, and then do the comparison.

Reply:
Will make the change 👍Thank you :)